service/src/main/java/bio/terra/pipelines/common/utils/PipelineVariableTypesEnum.java

```diff
@@ -165,9 +167,10 @@ public String validate(String fieldName, Object value) {
    * the value fails validation, return a descriptive error message in a String.
    *
    * @param fieldName - the name of the field being validated (used to construct error messages)
+   * @paramf fileSuffix - nullable suffix used to validate file types
    * @param value - the value to validate
    */
-  public abstract String validate(String fieldName, Object value);
+  public abstract String validate(String fieldName, String fileSuffix, Object value);
```

Comment:
hmm i think this signature might be better if we change it to 
```
public abstract String validate(PipelineVariableTypesEnum pipelineVariableTypesEnum, Object value);
```
and then inside the overriding functions we can grab the field name and the suffix if its needed.   What do you think?

Reply:
hm, the suffix isn't stored in the `PipelineVariableTypesEnum` object, it's in each individual `PipelineInputDefinition` object. if we're confident we're only ever going to call validate on inputs, then we could do 
`public abstract String validate(PipelineInputDefinition pipelineInputDefinition, Object value)`
and the rest like you suggest. what do you think?

Reply:
if we do pass in the PipelineInputDefinition directly for now, and later decide we want to run validate on pipeline outputs too, then `fileSuffix` will be added to the `BasePipelineVariableDefinition` and we should be able to amend the signature to take an implementation of `BasePipelineVariableDefinition`, right? i.e. doing this can be compatible with validating on outputs if we decide to do that in the future

Reply:
yup i copy pasted the wrong thign lol.  this is exactly right. PipelineInputDefinition for now and if we want to validate outputs (which we _shouldnt_ have to) then BasePipelineVariableDefinition with a file_suffix being added to the outputs table

Reply:
ok perfect. great idea, thanks!